app/src/main/java/com/example/benchproject/app/common/navigator/FragmentNavigator.kt

```diff
@@ -16,7 +17,9 @@ abstract class FragmentNavigator {
             this.fragment = null
     }
 
-    fun errorSnackBar(message: String) {
-        Snackbar.make(fragment!!.requireView(), message, Snackbar.LENGTH_SHORT).show()
+    fun errorSnackBar(message: String, lambda: () -> Unit) {
+        Snackbar.make(fragment!!.requireView(), message, Snackbar.LENGTH_LONG)
+            .setAction(fragment!!.getString(R.string.snackbarRetry)) { lambda() }
```

Comment:
1. Notice repetition of `fragment!!` - DRY rule
2. You're passing `Snackbar.LENGTH_LONG` - does it mean that after some time snack bar will disappear and user will lost a possibility to tap on "Retry"?

Reply:
1. I created a variable for fragment
2. I created a function for new snackbar without action